tests/QueryWithFiltersTest.php

```diff
@@ -0,0 +1,165 @@
+            ],
+            //GreaterThanHandler
+            [
+                new FilterGreaterThan('column', 1000),
+                '[column] > 1000',
+            ],
+            [
+                new FilterGreaterThan('column', new DateTime('2011-01-01T15:00:01')),
+                "[column] > '2011-01-01 15:00:01'",
+            ],
+            [
+                new FilterGreaterThanOrEqual('column', 3.5),
+                '[column] >= 3.5',
+            ],
+            [
+                new FilterLessThan('column', 10.7),
+                '[column] < 10.7',
+            ],
+            [
+                new FilterLessThanOrEqual('column', 100),
+                '[column] <= 100',
+            ],
+            [
+                new FilterIn('column', [10, 20.5, 30]),
+                '[column] IN (10, 20.5, 30)',
+            ],
+            //NotHandler equals
+            [
+                new FilterNotEquals('column', 40),
+                '[column] != 40',
+            ],
+            //LikeHandler
+            [
+                new FilterLike('column', 'foo'),
+                "[column] LIKE '%foo%'",
+            ],
+            [
+                (new FilterLike('column', 'foo'))->withoutStart(),
+                "[column] LIKE 'foo%'",
+            ],
+            [
+                (new FilterLike('column', 'foo'))->withoutEnd(),
+                "[column] LIKE '%foo'",
+            ],
+            [
+                (new FilterLike('column', 'foo'))->withoutBoth(),
+                "[column] LIKE 'foo'",
+            ],
+            [
+                new FilterLike('column', new Expression("CONCAT([[foo]] ->> 'bar', '%')")),
+                "[column] LIKE CONCAT([foo] ->> 'bar', '%')",
+            ],
+            //Array Or FilterLike
+            [
+                new FilterOrLike('column', ['foo', 'bar']),
+                "[column] LIKE '%foo%' OR [column] LIKE '%bar%'",
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider simpleDataProvider
+     */
+    public function testSimpleFilter(CompareFilter $filter, string $condition): void
+    {
+        $db = $this->getConnection();
+        $query = (new Query($db))
+            ->from('customer');
+
+        $dataReader = (new QueryDataReader($query))
+            ->withFilter($filter);
+
+        $expected = 'SELECT * FROM [customer] WHERE ' . $condition;
+
+        $this->assertSame(
+            $dataReader->getPreparedQuery()->createCommand()->getRawSql(),
+            $expected,
+        );
+    }
+
+    /**
+     * @dataProvider simpleDataProvider
+     */
+    public function testSimpleHaving(CompareFilter $having, string $condition): void
+    {
+        $db = $this->getConnection();
+        $query = (new Query($db))
+            ->from('customer');
+
+        $dataReader = (new QueryDataReader($query))
+            ->withHaving($having);
+
+        $expected = 'SELECT * FROM [customer] HAVING ' . $condition;
```

Comment:
Shouldn't it produce `{{%table name}}` instead? 


https://github.com/yiisoft/db/tree/master/docs/en#quote-table-and-column-names

Reply:
> Shouldn't it produce `{{%table name}}` instead?
> 
> https://github.com/yiisoft/db/tree/master/docs/en#quote-table-and-column-names

I just copy test support from yii-db test (small cheating) in order to write them quickly

Reply:
> Shouldn't it produce `{{%table name}}` instead?
> 
> https://github.com/yiisoft/db/tree/master/docs/en#quote-table-and-column-names

SqlLite allow square brackets for quote column or table name